.github/workflows/label-issue-open.yml

```diff
@@ -0,0 +1,75 @@
+name: Add New Issue/Pull Labels
+
+on:
+  pull_request_target:
+    types: opened
+  issues:
+    types: opened
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+  
+permissions: {}
+
+jobs:
+  triage:
+    name: Add awaiting triage
+    if: github.event.issue
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: read
+      issues: write
+    steps:
+      - uses: actions/checkout@v3
+      # We want to delay the labeling of the issue so that the author has a change to add labels after issue creation.
+      - name: 'Delay Labeling'
+        run: sleep 3m
+      # Make sure that the latest issue is pulled from the database rather than relying on the payload. This is
+      # because the payload won't include any labels that were added after the issue was created.
+      - uses: actions/github-script@v6
+        id: latest-issue
+        with:
+          script: |
+            const issue = await github.rest.issues.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number
+            });
+            core.setOutput('hasLabels', issue.data.labels.length > 0);
+      - uses: actions-ecosystem/action-add-labels@v1
+        if: ${{ steps.latest-issue.outputs.hasLabels == 'false'}}
+        with:
+          labels: 'status: awaiting triage'
+
+  community:
+    name: Add community contribution label
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: read
+      pull-requests: write
+      issues: write
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c
+
+      - name: Install Octokit
+        run: npm --prefix .github/workflows/scripts install @octokit/action
+
+      - name: Install Actions Core
+        run: npm --prefix .github/workflows/scripts install @actions/core
```

Comment:
If you use `actions/github-script` you can streamline this and remove these installation steps. You can still use internal scripts too, see [this section in the readme](https://github.com/actions/github-script#run-a-separate-file). I used this in `ci.yml` to separate some the logic from GitHub to make it easier to test. In theory this lets you write nice automated tests and at some point we should clean up `.github/workflows/actions`.

Reply:
@ObliviousHarmony Thanks for the suggestion. I've updated the PR.